src/relay/backend/aot_executor_codegen.cc

```diff
@@ -177,6 +179,12 @@ class AOTOnDemandAllocator : public transform::DeviceAwareExprVisitor {
       for (auto sid : sinfo->storage_ids) {
         return_ids_.push_back(sid);
       }
+      return_ttypes_.clear();
+      auto ttypes = FlattenTupleType(e->checked_type());
+      return_ttypes_.reserve(ttypes.size());
+      for (auto ttype : ttypes) {
+        return_ttypes_.push_back(ttype);
+      }
```

Comment:
All of this is equivalent to `return_ttypes_ = FlattenTupleType(e->checked_type());`.

Reply:
cc @areusch